services/viva-ms/lambdas/syncOfficers.js

```diff
@@ -31,60 +30,49 @@ export async function main(event, context) {
   );
   if (vadaMyPagesError) {
     log.error(
-      'vada mypages error',
+      'Could not get Viva applicant administrator(s)',
       context.awsRequestId,
       'service-viva-ms-syncOfficers-001',
       vadaMyPagesError
     );
 
-    return null;
+    return false;
   }
 
   const { officer } = vadaMyPagesResponse;
   const vivaAdministrators = parseVivaOfficers(officer);
 
   if (deepEqual(vivaAdministrators, administrators)) {
-    return null;
+    return false;
   }
 
-  const TableName = config.cases.tableName;
-
-  const UpdateExpression = 'SET details.administrators = :newAdministrators';
-  const ExpressionAttributeValues = { ':newAdministrators': vivaAdministrators };
-
-  const dynamoDbParams = {
-    TableName,
-    Key: {
-      PK,
-      SK,
-    },
-    UpdateExpression,
-    ExpressionAttributeValues,
-    ReturnValues: 'UPDATED_NEW',
-  };
-
-  const [updateError] = await to(sendUpdateRequest(dynamoDbParams));
+  const [updateError] = await to(sendUpdateRequest({ PK, SK }, vivaAdministrators));
   if (updateError) {
     log.error(
-      'Update request error',
+      'Could not update case applicant administrator(s)',
       context.awsRequestId,
       'service-viva-ms-syncOfficers-002',
       updateError
     );
 
-    return null;
+    return false;
   }
 
   return true;
 }
 
-async function sendUpdateRequest(params) {
-  const [error, result] = await to(dynamoDb.call('update', params));
-  if (error) {
-    throwError(500, error.message);
-  }
+async function sendUpdateRequest(keys, newAdministrators) {
+  const TableName = config.cases.tableName;
+
+  const dynamoDbParams = {
+    TableName,
+    Key: keys,
```

Comment:
What are the keys? Suggestion is that we destruct the keys argument so that we know what keys we are passing.
```JS
const {PK, SK} = keys
const params = {
  ...,
  Key: {PK, SK},
  ...
}
```

Reply:
This is how we do it elsewhere? But yeah... good point.

Reply:
Let's rewrite our past then ;)

Reply:
Will be fixed in future task